solutions/CF Garland.mdx

```diff
@@ -2,10 +2,78 @@
 id: cf-garland
 source: CF
 title: An impassioned circulation of affection
-author: Neo Wang
+author: Neo Wang, Qi Wang
 ---
 
-[Official Editorial](https://codeforces.com/blog/entry/52449)
+[Official DP Editorial](https://codeforces.com/blog/entry/52449)
+
+## Solution (Two Pointer)
+
+**Time Complexity:** $\mathcal{O}(nq)$
+
+We can keep a sliding window of maximum consecutive runs of that letter with a maximum of $m$ letters exchanged.
```

Comment:
Probably should elaborate.

Reply:
Also sliding window or running sum?

Reply:
Woops nvm

Reply:
For each query we can keep a sliding window where no more than $a$ letters in the window are not $b$

something more like this???
idk

Reply:
oh the variables in your soln are different from those in the problem